cEP-0009.md

```diff
@@ -0,0 +1,181 @@
+| Metadata     |                                                          |
+| ------------ |----------------------------------------------------------|
+| cEP          | 9                                                        |
+| Version      | 0.1                                                      |
+| Title        | Extract and utilize information from project files       |
+| Authors      | Satwik Kansal <satwikkansal@gmail.com>                   |
+| Status       | Proposed                                                 |
+| Type         | Feature                                                  |
+
+## Abstract
+
+There are many kind of project files like ".editorconfig", "Gruntfile.js",
+etc. that contain meta-information that can used by tools like
+coala-quickstart to provide better analysis. This cEP proposes a mechnaism to
+represent and extract information from project files and an interface to utilize the extracted information.
+
+## Proposed Approach
+
+### 1. Implementation of the “Info” class
+
+A subclass of `Info` class will represent a specific kind of information. For example, the class `IgnorePathInfo` may represent “ignored file paths”.
+
+```py
+class Info:
+    description = "Some information"
+    # for validation
+    value_type = "data-type of the information value, can also be an ``Enum``"
+                 "of possible choices."
+
+    def __init__(self,
+                 source,
+                 value: value_type):
+        self.source = source
+        self.value = value
+        self.name = self.__class__.__name__
+        # optional metadata
+        self.scope = "bear-specific" #other possible values may be "global"
+
+    def get_value(self):
+        return self.value
+```
+
+
+### 2. Implementation of “InfoExtractor” class
+Any derived class of the `InfoExtractor` abstract class consists of:
+- Methods for parsing the project file.
+- Methods to find information in the parsed file.
+- A collection of `Info` objects representing the information extracted from a particular kind of project file.
+- Files to parse and extract information from.
+
+```py
+class InfoExtractor:
+    # Kinds of files supported by the InfoExtractor
+    supported_files = [**]
+
+    # Meta-data
+    supported_information_kinds = []
+
+    def __init__(self,
+                 target_files,
+                 project_directory):
+        self.target_files = target_files
+        self.directory = directory
+        self.information = dict()
+
+    def parse_file(self, filename):
+        """
+        Parses the given file and returns the parsed file.
+        """
+        raise NotImplementedError
+
+    def find_information(self, fname, parsed_file):
+        """
+        Returns a collection of ``Info`` instances.
+        """
+        raise NotImplementedError
+
+    def _add_info(self, info:Info):
+        """
+        Method to organize and add the supplied information in self.information attribute.
+        """
+        pass
```

Comment:
I have some confusion regarding the structure of the stored information inside the `InfoExtractor`
This is what I've in mind 

- A list of `Info` objects.

or

```
{
	info_name_1: [],
	info_name_2: []
}
```
or

```
{
	info_name_1: {
		filename_1: value1
	}
}
```

or 

```
{
	filename_1: {
		info_name_1: value1,
		info_name_2: value2
	}
}
```